contracts/token/RarePizzas.sol

```diff
@@ -0,0 +1,265 @@
+
+    string public constant _uriBase = 'ipfs://';
+
+    string private _contractURI;
+
+    // Other contracts this contract interacts with
+    IOrderAPIConsumer internal _orderAPIClient;
+    IRarePizzasBox internal _rarePizzasBoxContract;
+
+    CountersUpgradeable.Counter public _minted_pizza_count;
+    CountersUpgradeable.Counter public _purchased_pizza_count;
+
+    event RarePizzasBoxContractUpdated(address previous, address current);
+    event OrderAPIClientUpdated(address previous, address current);
+    event InternalArtworkAssigned(uint256 tokenId, bytes32 artworkURI);
+
+    // A collection of Box Token Id's that have been redeemed
+    mapping(uint256 => address) internal _redeemedBoxTokenAddress;
+
+    // A collection of all of the pizza artwork IPFS hashes
+    mapping(uint256 => bytes32) internal _tokenPizzaArtworkURIs;
+
+    // A collection of render jobs associated with the requestor
+    mapping(bytes32 => address) internal _renderRequests;
+
+    // A collection of render jobs associated with the box token id
+    mapping(bytes32 => uint256) internal _renderTokenIds;
+
+    // END V1 Variables
+
+    function initialize(address rarePizzasBoxContract) public initializer {
+        __Ownable_init();
+        __ReentrancyGuard_init();
+        __ERC721_init('Rare Pizzas', 'PIZZA');
+
+        if (rarePizzasBoxContract != address(0)) {
+            _rarePizzasBoxContract = IRarePizzasBox(rarePizzasBoxContract);
+        }
+
+        _contractURI = 'https://raw.githubusercontent.com/PizzaDAO/pizza-smartcontract/master/data/opensea_pizza_metadata.mainnet.json';
+    }
+
+    // IOpenSeaCompatible
+    function contractURI() public view virtual override returns (string memory) {
+        // Metadata provided via github link so that it can be updated or modified
+        return _contractURI;
+    }
+
+    // IRarePizzas
+
+    function isRedeemed(uint256 boxTokenId) public view override returns (bool) {
+        return _redeemedBoxTokenAddress[boxTokenId] != address(0);
+    }
+
+    function redeemRarePizzasBox(uint256 boxTokenId) public override nonReentrant {
+        require(_msgSender() == _rarePizzasBoxContract.ownerOf(boxTokenId), 'caller must own box');
+        _redeemRarePizzasBox(_msgSender(), boxTokenId);
+    }
+
+    // IRarePizzasBox overrides
+
+    function getBitcoinPriceInWei() public view virtual override returns (uint256) {
+        return _rarePizzasBoxContract.getBitcoinPriceInWei();
+    }
+
+    function getPrice() public view virtual override returns (uint256) {
+        return _rarePizzasBoxContract.getPriceInWei();
+    }
+
+    function getPriceInWei() public view virtual override returns (uint256) {
+        return _rarePizzasBoxContract.getPriceInWei();
+    }
+
+    function maxSupply() public view virtual override returns (uint256) {
+        return MAX_TOKEN_SUPPLY;
+    }
+
+    function purchase() public payable virtual override nonReentrant {
```

Comment:
this one can probably be removed. we dont want people trying to purchase on this contract i dont think